app/uk/gov/hmrc/apinotificationpull/services/ApiNotificationQueueService.scala

```diff
@@ -19,15 +19,28 @@ package uk.gov.hmrc.apinotificationpull.services
 import javax.inject.Inject
 
 import uk.gov.hmrc.apinotificationpull.connectors.ApiNotificationQueueConnector
-import uk.gov.hmrc.apinotificationpull.model.Notifications
+import uk.gov.hmrc.apinotificationpull.model.{Notification, Notifications}
 import uk.gov.hmrc.http.HeaderCarrier
 
 import scala.concurrent.Future
+import scala.concurrent.ExecutionContext.Implicits.global
 
 class ApiNotificationQueueService @Inject()(apiNotificationQueueConnector: ApiNotificationQueueConnector) {
 
   def getNotifications()(implicit hc: HeaderCarrier): Future[Notifications] = {
     apiNotificationQueueConnector.getNotifications()
   }
 
+  def getAndRemoveNotification(notificationId: String)(implicit hc: HeaderCarrier): Future[Option[Notification]] = {
```

Comment:
Could you improve the implementation of `getAndRemoveNotification`? We can make it more clean and coincise